src/h264/H26xPacketizer.cpp

```diff
@@ -54,22 +53,23 @@ void H26xPacketizer::EmitNal(VideoFrame& frame, BufferReader nal, std::string& f
 	//Skip payload nal header
 	nal.Skip(naluHeaderSize);
 	pos += naluHeaderSize;
-	//Split it
-	while (nal.GetLeft())
+	
+	auto numPackets = nal.GetLeft() / (RTPPAYLOADSIZE - fuPrefixSize) + 1;
```

Comment:
wouldn't this give an incorrect packet count if mod is 0? i.e. if `nal.GetLeft()` is `(RTPPAYLOADSIZE - fuPrefixSize)`

`auto numPackets = (RTPPAYLOADSIZE - fuPrefixSize) / (RTPPAYLOADSIZE - fuPrefixSize) + 1; ` is 2 an not 1.


```suggestion
	auto numPackets = std::ceil(nal.GetLeft() / (RTPPAYLOADSIZE - fuPrefixSize));
```

Reply:
could you add a test for that as well?

Reply:
if `nal.GetLeft()` is `(RTPPAYLOADSIZE - fuPrefixSize)` then it can be fit into one single packet right? which would not need FU-A anymore I think. I can definitely add a test with nalUnitSize == 1200 bytes which is the min bytes for FU-A mode.

Reply:
what I mean is that the code above will fail if mod=0